### PR TITLE
 Oracle SQL Sessions Assignments-1

### PR DESCRIPTION
**Pull request by Bappy-kumar**
This is my Assignment-1

### CODE REVIEW

**Review by Bappy-kumar — COMMENTED** [review]
Hello this is my Assignment

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 1]
```diff
@@ -0,0 +1,102 @@
+1. select count(*) from emp;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:2]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+2. select count(*) from dept;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:4]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+3. select * from emp;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:6]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+4. select * from dept;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:8]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+5. select sum(sal)
+   from emp;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:11]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+3. select * from emp;
+
+4. select * from dept;
+
+5. select sum(sal)
+   from emp;
+   
+6. select sum(comm)
+   from emp;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:14]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+5. select sum(sal)
+   from emp;
+   
+6. select sum(comm)
+   from emp;
+   
+7. select distinct job from emp
+   where comm is not null;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:17]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+   from emp;
+   
+6. select sum(comm)
+   from emp;
+   
+7. select distinct job from emp
+   where comm is not null;
+   
+8. select sysdate from dual; 
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:19]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+9. select avg(sal)
+   from emp;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:22]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+   
+8. select sysdate from dual; 
+
+9. select avg(sal)
+   from emp;
+   
+10. select deptno,count(*) "no of emp" 
+    from emp 
+    group by deptno;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:26]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+   from emp;
+   
+10. select deptno,count(*) "no of emp" 
+    from emp 
+    group by deptno;
+    
+ 11. select deptno,sum(sal) "no of emp" 
+     from emp 
+     group by deptno;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:30]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 12]
```diff
@@ -0,0 +1,102 @@
...
+ 12. select job, count(*) 
+      from emp
+      group by job;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:34]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 13]
```diff
@@ -0,0 +1,102 @@
...
+     group by deptno;
+
+ 12. select job, count(*) 
+      from emp
+      group by job;
+      
+  13. select job, avg(sal) 
+      from emp
+      group by job;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:38]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 14]
```diff
@@ -0,0 +1,102 @@
...
+      
+  13. select job, avg(sal) 
+      from emp
+      group by job;
+      
+  14. select to_char(hiredate,'DD')
+      as hire_date,to_char(hiredate,'MM') 
+      as hire_month,to_char(hiredate,'YYYY') 
+      as hire_year from emp;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:43]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 15]
```diff
@@ -0,0 +1,102 @@
...
+      
+  14. select to_char(hiredate,'DD')
+      as hire_date,to_char(hiredate,'MM') 
+      as hire_month,to_char(hiredate,'YYYY') 
+      as hire_year from emp;
+      
+15.  select * 
+     from emp 
+     order by deptno;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:47]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 16]
```diff
@@ -0,0 +1,102 @@
...
+      as hire_year from emp;
+      
+15.  select * 
+     from emp 
+     order by deptno;
+     
+16.   select * 
+      from emp 
+      order by job;  
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:51]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 17]
```diff
@@ -0,0 +1,102 @@
...
+     order by deptno;
+     
+16.   select * 
+      from emp 
+      order by job;  
+      
+17.   select * 
+      from emp 
+      order by sal desc;  
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:55]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 18]
```diff
@@ -0,0 +1,102 @@
...
+      order by job;  
+      
+17.   select * 
+      from emp 
+      order by sal desc;  
+      
+ 18.   select * 
+       from emp 
+       order by deptno,sal desc;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:59]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+      order by sal desc;  
+      
+ 18.   select * 
+       from emp 
+       order by deptno,sal desc;
+       
+ 19.   select count(ename) 
+      "name with 6 characters" from emp 
+       where Ename like '______%';
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:63]: not ok.  if we ask for say length of name to be 3, then ___ change.

select * from emp where length(ename)=6;

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+       from emp 
+       order by deptno,sal desc;
+       
+ 19.   select count(ename) 
+      "name with 6 characters" from emp 
+       where Ename like '______%';
+    
+ 20.   select max(sal),min(sal) 
+       from emp;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:66]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+ 20.   select max(sal),min(sal) 
+       from emp;
+    
+ 21.   select deptno,max(sal) 
+       "max",min(sal) 
+       "min",avg(sal) 
+       "avg",sum(sal) "sum" 
+        from emp 
+        group by deptno;
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:73]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 22]
```diff
@@ -0,0 +1,102 @@
...
+       "max",min(sal) 
+       "min",avg(sal) 
+       "avg",sum(sal) "sum" 
+        from emp 
+        group by deptno;
+    
+ 22.    select * 
+        from emp 
+        order by hiredate;  
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:77]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 23]
```diff
@@ -0,0 +1,102 @@
...
+    
+ 22.    select * 
+        from emp 
+        order by hiredate;  
+        
+23.     select *
+        from emp
+        where hiredate = (select max(hiredate) 
+        from emp);  
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:82]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 24]
```diff
@@ -0,0 +1,102 @@
...
+23.     select *
+        from emp
+        where hiredate = (select max(hiredate) 
+        from emp);  
+      
+24.      select *
+         from emp
+         where hiredate = (select min(hiredate) 
+         from emp);  
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:87]: ok

File: Assignments/Bappy/Oracle SQL Sessions Assignments-1 [hunk 25]
```diff
@@ -0,0 +1,102 @@
...
+         order by hiredate desc,deptno;         
+
+26.      select empno,ename,sal
+	     from emp
+	     where sal >= (select (avg(sal))from emp);
+ 
+ 27.     select empno,ename,sal
+	     from emp
+         where sal <= (select (avg(sal))from emp);
+
```
Comment (raviproengage) [inline Assignments/Bappy/Oracle SQL Sessions Assignments-1:99]: ok

**Review by raviproengage — COMMENTED** [review]
Good Submission. Keep it up!
but where are queries for questions # 29 & # 30?

**Comment by raviproengage** [comment]
ok

**Pull request closed without merging**